src/app/components/profile-edit/profile-edit.service.ts

```diff
@@ -62,4 +64,27 @@ export class ProfileEditService implements OnDestroy {
                 );
         }
     }
+    setUrl(url: string, deletePreviousPath: string): any {
+        this.deletePreviousPath = deletePreviousPath;
+        return (this.newUrl = url);
+    }
+    getUrl() {
+        return this.newUrl;
+    }
+    delPreviousUrl() {
+        if (this.deletePath === ' ') {
+            this.deletePath = this.deletePreviousPath;
+        } else if (this.deletePath !== this.deletePreviousPath) {
+            const path = `${this.deletePath}`;
+            const delRef = this.storage.ref(path);
+            delRef
+                .delete()
+                .toPromise()
```

Comment:
why?

Reply:
what are you talking about? I tried to create del func to clean about my self. Because in fireBase we have limit on content size, besides when i code my component i have created a lot of trash, and in firebase bad possibilities to delete my trash.

Reply:
what is the reason of converting to empty promise and having empty bodies?

Reply:
hard to explain. But without this construction doesn't work.